### PR TITLE
replace tcp by http request for using http_proxy

### PR DESCRIPTION
**Pull request by kingofstormland**
replace tcp by http request for using http_proxy

### CODE REVIEW

File: pkg/platform/provider/imported/cluster/provider.go [hunk 1]
```diff
@@ -110,6 +111,28 @@ func (p *Provider) ValidateCredential(cluster clusterprovider.InternalCluster) (
 	return allErrs, nil
 }
 
+func dialBySimpleHttp(host string, port int32, timeout time.Duration) (bool, error){
+	client := &http.Client{
+		Timeout: timeout,
+	}
+
+	url := "http://"+host+":"+strconv.Itoa(int(port))
```
Comment (choujimmy) [inline pkg/platform/provider/imported/cluster/provider.go]: don't use http.Client.Timeout, use net.Dialer.Timeout

File: pkg/platform/provider/imported/cluster/provider.go [hunk 2]
```diff
@@ -110,6 +111,28 @@ func (p *Provider) ValidateCredential(cluster clusterprovider.InternalCluster) (
...
+	fmt.Println(url)
```
Comment (choujimmy) [inline pkg/platform/provider/imported/cluster/provider.go]: plz remove this

File: pkg/platform/provider/imported/cluster/provider.go (hunk 1 above)
Comment (choujimmy) [inline pkg/platform/provider/imported/cluster/provider.go]: The kube-apiserver has the https service enabled by default. Using the http protocol to detect will cause a tls handshake error to the normal apiserver.

**Comment by QianChenglong** [comment]
Consider move this validation to util package like https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

**Pull request merged**